From: Devaraj Rangasamy <dev@ti.com>
Date: Fri, 22 Apr 2011 22:59:41 +0000 (+0530)
Subject: DSS: disable HDMI overlays on suspend
X-Git-Tag: android-2.6.35-2.3-omap4.13.1~16
X-Git-Url: http://git.omapzoom.org/?p=kernel%2Fomap.git;a=commitdiff_plain;h=5a0723b9cf5ca80fabe64e44a5d1391df9a87711

DSS: disable HDMI overlays on suspend

On suspend call, HDMI driver should disable its panel manager.
Not doing so, leaves with active configuration on HDMI-force suspend
scenairo, leaving no DSS MSTBY and no core/dss domain transitions.
This patch fixes the break.

Change-Id: Ida281eab6609fe4e8c47127e6491e2cde4ba110d
Signed-off-by: Devaraj Rangasamy <dev@ti.com>
---

diff --git a/drivers/video/omap2/dss/hdmi.c b/drivers/video/omap2/dss/hdmi.c
index b5a97dc..6066c9a 100644
--- a/drivers/video/omap2/dss/hdmi.c
+++ b/drivers/video/omap2/dss/hdmi.c
@@ -1715,6 +1715,7 @@ static int hdmi_set_power(struct omap_dss_device *dssdev,
 		} else if (power_need == HDMI_POWER_MIN) {
 			r = hdmi_min_enable();
 		} else {
+                        dssdev->manager->disable(dssdev->manager);
 			omap_dss_stop_device(dssdev);
 			hdmi_power_off(dssdev);
 		}